src/tox.ini

```diff
@@ -239,6 +239,66 @@ description =
+#    433        PLR5501 [*] Use `elif` instead of `else` then `if`, to reduce indentation
+#    428        PLR0911 [ ] Too many return statements (10 > 6)
+#    404        E731    [*] Do not assign a `lambda` expression, use a `def`
+#    351        F405    [ ] `ComplexField` may be undefined, or defined from star imports
+#    306        PLR1714 [*] Consider merging multiple comparisons. Use a `set` if the elements are hashable.
+#    236        F403    [ ] `from .abelian_gps.all import *` used; unable to detect undefined names
+#    116        PLR0402 [*] Use `from matplotlib import cm` in lieu of alias
+#    111        PLW0603 [ ] Using the global statement to update `AA_0` is discouraged
+#     78        F841    [*] Local variable `B` is assigned to but never used
+#     64        E713    [*] Test for membership should be `not in`
+#     48        PLW0602 [ ] Using global for `D` but no assignment is done
+#     33        PLR1711 [*] Useless `return` statement at end of function
+#     24        E714    [*] Test for object identity should be `is not`
+#     20        PLR1701 [*] Merge `isinstance` calls
+#     17        PLW3301 [ ] Nested `max` calls can be flattened
+#     16        PLW1510 [*] `subprocess.run` without explicit `check` argument
+#     14        E721    [ ] Do not compare types, use `isinstance()`
+#     14        PLW0120 [*] `else` clause on loop without a `break` statement; remove the `else` and dedent its contents
+#     12        F811    [*] Redefinition of unused `CompleteDiscreteValuationRings` from line 49
+#      8        PLC0414 [*] Import alias does not rename original package
+#      7        E743    [ ] Ambiguous function name: `I`
+#      7        PLE0101 [ ] Explicit return in `__init__`
+#      7        PLR0124 [ ] Name compared with itself, consider replacing `a == a`
+#      5        PLW0127 [ ] Self-assignment of variable `a`
+#      4        F541    [*] f-string without any placeholders
+#      4        PLW1508 [ ] Invalid type for environment variable default; expected `str` or `None`
+#      3        PLC3002 [ ] Lambda expression called directly. Execute the expression inline instead.
+#      2        E742    [ ] Ambiguous class name: `I`
+#      2        PLE0302 [ ] The special method `__len__` expects 1 parameter, 3 were given
+#      2        PLW0129 [ ] Asserting on a non-empty string literal will always pass
+#      1        F402    [ ] Import `factor` from line 259 shadowed by loop variable
+#      1        PLC0208 [*] Use a sequence type instead of a `set` when iterating over values
+#
+commands = ruff --ignore I001,PLR2004,F401,E741,F821,PLR0912,PLR0913,E402,PLR0915,PLW2901,PLR5501,PLR0911,E731,F405,PLR1714,F403,PLR0402,PLW0603,E713,F841,PLW0602,E714,PLR1711,PLR1701,PLW3301,E721,PLW1510,F811,PLW0120,PLC0414,E743,F541,PLE0101,PLR0124,PLW0127,PLW1508,PLC3002,E742,PLE0302,F402,PLC0208,PLW0129 {posargs:{toxinidir}/sage/}
+
```

Comment:
How about making the list in line 300 is ordered the same with the list above, for easy maintenance?

Reply:
Good idea, done in c97451dc3e4